service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -190,8 +223,267 @@ public void prepareCustomDirections(Map<String, Object> data) {
         }
     }
 
-    public Map<String, Object> generateCMOTemplateData(Map<String, Object> caseData) {
```

Comment:
I would probably create this piece of code in separate service, adding document generation template preparation in the same place as logic around CMO object handling baloons it to enormous size. What do you think?

Reply:
That would probably be a nice idea

Reply:
Created a new set of classes to clean this up as discussed